gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/YarnService.java

```diff
@@ -558,8 +556,11 @@ protected ByteBuffer getSecurityTokens() throws IOException {
   protected String buildContainerCommand(Container container, String helixParticipantId, String helixInstanceTag) {
     long allocationRequestId = container.getAllocationRequestId();
     // Using getOrDefault for backward-compatibility with containers that don't have allocationRequestId set
-    WorkerProfile workerProfile = this.workerProfileByAllocationRequestId.getOrDefault(allocationRequestId,
-        this.defaultWorkerProfile);
+    WorkerProfile workerProfile = Optional.fromNullable(this.workerProfileByAllocationRequestId.get(allocationRequestId))
+        .or(() -> {
+          LOGGER.warn("No Worker Profile found for {} ... falling back... ", allocationRequestId);
```

Comment:
sorry, I meant to suggest we fill in the ellipses, maybe:
```
"No Worker Profile found for {}, so falling back to default"
```

I suppose to be truly robust it would look like
```
    WorkerProfile workerProfile = Optional.fromNullable(this.workerProfileByAllocationRequestId.get(allocationRequestId))
    .or(() -> {
      LOGGER.warn("No Worker Profile found for {}, so falling back to default", allocationRequestId);
      return this.workerProfileByAllocationRequestId.computeIfAbsent(DEFAULT_ALLOCATION_REQUEST_ID, k -> {
        LOGGER.warn("WARNING: (LIKELY) UNEXPECTED CONCURRENCY: No Worker Profile even yet mapped to the default allocation request ID {} - creating one now", DEFAULT_ALLOCATION_REQUEST_ID);
        return new WorkerProfile(this.config);
      });
    });
```

Reply:
Thanks for this suggestion, adding the same